storage.bs

```diff
@@ -94,37 +100,251 @@ function retrieveNextChunk(nextChunkInfo) {
+being replaced due to a cross-origin opener policy.
+
+
+<h3 id=storage-endpoints>Storage endpoints</h3>
+
+<p>A <dfn export>storage endpoint</dfn> is a storage or session storage API that uses the
+infrastructure defined by this standard to keep track of its storage needs.
+
+<p>A <a>storage endpoint</a> has an <dfn for="storage endpoint">identifier</dfn>, which is a
+<a>storage identifier</a>.
+
+<p>A <a>storage endpoint</a> also has <dfn for="storage endpoint">types</dfn>, which is a
+<a for=/>set</a> of <a>storage types</a>.
+
+<p>A <dfn>storage identifier</dfn> is an <a for=/>ASCII string</a>.
+
+<hr>
+
+<p>The <dfn>registered storage endpoints</dfn> are a <a for=/>set</a> of <a>storage endpoints</a>
+defined by the following table:
+
+<table>
+ <tr>
+  <th><a>Storage identifier</a>
+  <th><a>Storage types</a>
+ <tr>
+  <td>"<code>caches</code>"
+  <td>« "<code>storage</code>" »
+ <tr>
+  <td>"<code>indexedDB</code>"
+  <td>« "<code>storage</code>" »
+ <tr>
+  <td>"<code>localStorage</code>"
+  <td>« "<code>storage</code>" »
+ <tr>
+  <td>"<code>serviceWorkerRegistrations</code>"
+  <td>« "<code>storage</code>" »
+ <tr>
+  <td>"<code>sessionStorage</code>"
+  <td>« "<code>session-storage</code>" »
+</table>
+
+
+
+<h3 id=buckets oldids=boxes>Storage buckets</h3>
+
+<p>An <dfn>abstract storage bucket</dfn> is a place for <a>storage endpoints</a> to store data.
+Whenever an <a>abstract storage bucket</a> is cleared by the user agent, it must be cleared in its
+entirety.
+<!-- This is probably better defined in some kind of storage lifetime section. -->
+
+<p>An <a>abstract storage bucket</a> has a
+<dfn for="abstract storage bucket,storage bucket,session storage bucket">map</dfn> of
+<a>storage identifiers</a> to <a for="abstract storage bucket">areas</a>.
 
-<p><a>Storage</a> consists of zero or more <dfn export id=site-storage-unit>storage units</dfn>.
+<p>An <dfn for="abstract storage bucket,storage bucket,session storage bucket">area</dfn> is a part
+of an <a>abstract storage bucket</a> carved out for a <a>storage endpoint</a>. An
+<a for="abstract storage bucket">area</a> has a
+<dfn for="abstract storage bucket/area,storage bucket/area,session storage bucket/area">map</dfn>,
+which is initially an empty <a for=/>map</a>. An <a for="abstract storage bucket">area</a> also has
+a
+<dfn for="abstract storage bucket/area,storage bucket/area,session storage bucket/area">proxy map reference set</dfn>,
+which is initially an empty <a for=/>set</a>.
 
-<p>Each <a for=/>origin</a> has an associated <a>storage unit</a>. A <a>storage unit</a> contains a
-single <dfn export id=bucket oldids=box>bucket</dfn>. [[HTML]]
+<p>How an <a>abstract storage bucket</a>'s <a for="abstract storage bucket">area</a>'s
+<a for="abstract storage bucket/area">map</a> is stored is <a>implementation-defined</a>. How it is
+made available across <a>agent</a> or even <a>agent cluster</a> boundaries is
+<a>implementation-defined</a>.
```

Comment:
This is a bit confusing. Upon re-reading it's clear, but at first I misread it as "_Whether_ it is made available across agent or event agent cluster boundaries is implementation-defined".

I'd rephrase this section as something like

> An area's map is where the actual data meant to be stored lives. User agents are expected to store this data, and make it available across agents and agent clusters, in an implementation-defined manner, so that when this specification (or other specifications?) access the contents of the map, they are available.